### PR TITLE
fix compilation issues

### PR DESCRIPTION
**Pull request by qindazhu**
@danpovey, note that I comment `MaxPerSublist` for now as it's the only one interface in `ops.h`  that uses `ragged.h` , I will try to implement some simple algorithms about `raggedArray` later to compile and test `ragged.h`

### CODE REVIEW

File: k2/csrc/cuda/array.h [hunk 1]
```diff
@@ -95,24 +92,24 @@ class Array1 {
...
+    ContextPtr c = Context();
+    assert(c->IsCompatible(indexes.GetContext()));
     int32_t ret_dim = indexes.Dim();
     Array1<T> ans(c, ret_dim);
     const T *this_data = Data();
     T *ans_data = ans.Data();
     int32_t *indexes_data = indexes.Data();
-    auto lambda_copy_elems = [=] __host__ __device__ (int32_t i) -> void {
-       ans_data[i] = this_data[indexes_data[i]];
+    auto lambda_copy_elems = [=] __host__ __device__(int32_t i) -> void {
```
Comment (csukuangfj) [inline k2/csrc/cuda/array.h:105]: It would be better if you specify explicitly what to be captured in the lambda.

Comment (qindazhu) [reply k2/csrc/cuda/array.h:105]: Sure, we will definitely do this when implementing other methods in `array.h`, the code from @danpovey now in `array.h` is just to delcare the interface

**Comment by danpovey** [comment]
Fangjun: most of the lambdas are going to have a lot of arguments, and the
idea of listing everything doesn't seem to scale adequately.
That's why I have decided to not list the captures.

File: k2/csrc/cuda/array.h (hunk 1 above)
Comment (csukuangfj) [reply k2/csrc/cuda/array.h:105]: The signature of this function says it returns `Array1`, but this function returns nothing.

Comment (danpovey) [reply k2/csrc/cuda/array.h:105]: ah, it should return `ans`

**Pull request merged**